bin/remote-prover/src/shared/mod.rs

```diff
@@ -0,0 +1,6 @@
+//! Shared types and utilities used by both client and server
+
+pub mod error;
+
+// Re-export commonly used items
+pub use error::*;
```

Comment:
Seems like the only think under this module is `errors`. Maybe this should be just `errors` module? (and then we can get rid of `shared` module).